analyzer/lib/concepts/where.rb

```diff
@@ -30,27 +34,43 @@ def initialize(name, description, specs: [])
       # @param [Position] position the position between target and atom
       # TODO: rspec
       def raw_position(target, spec_atom, position)
-        @raw_positions << [target, spec_atom, position]
+        @links[target] ||= []
+        @links[target] << [spec_atom, position]
       end
 
       # Swaps dependent specific spec
       # @param [SpecificSpec] from the spec which will be replaced
       # @param [SpecificSpec] to the spec to which swap will produce
+      # TODO: rspec
       def swap_source(from, to)
         return if from == to
         return unless @specs.delete(from)
         @specs << to
 
-        @raw_positions = @raw_positions.map do |target, spec_atom, position|
-          [target, swap(spec_atom, from, to), position]
+        @links = dup_graph(@links) do |v|
+          v.is_a?(Symbol) ? v : swap(v, from, to)
         end
+
         parents.each { |parent| parent.swap_source(from, to) }
       end
 
+      # Swaps atoms
+      # @param [Spec | SpecificSpec | VeiledSpec] spec the atom of which will be
+      #   swapped
+      # @param [Atom | AtomReference | SpecificAtom | VeiledAtom] from the used atom
+      # @param [Atom | AtomReference | SpecificAtom | VeiledAtom] to the new atom
+      # TODO: rspec
+      def swap_atom(spec, from, to)
```

Comment:
Unused method argument - `spec`. If it's necessary, use `_` or `_spec` as an argument name to indicate that it won't be used.